frontend/app/helpers/digital_object_helper.rb

```diff
@@ -0,0 +1,70 @@
+# Maps desired resource fields to new digital object instance fields as specified in ANW-1615
+module DigitalObjectHelper
+
+  def map_record_fields_to_digital_object(record)
+    if (record['jsonmodel_type'] == 'resource' or record['jsonmodel_type'] == 'accession')
+      record_hash = (record.class == Hash ? record : record.to_hash)
+    end
+
+    if not record_hash
+      raise ArgumentError.new("not a valid resource or accession JSONModel or hash")
+    end
+
+    copy_fields = ['title', 'dates', 'lang_materials']
+    processed_fields = record_hash.clone.keep_if {|k, v| copy_fields.include? k }
+    def cleanup!(data)
+      if data.is_a? Array
+        data.each {|item| cleanup!(item)}
+      end
+      if data.is_a? Hash
+        delete_fields = ['lock_version', 'created_by', 'last_modified_by', 'create_time', 'system_mtime', 'user_mtime']
+        data.delete_if {|k, v| delete_fields.include? k }
+        data.each_value {|v| cleanup!(v)}
+      end
+    end
+    cleanup!(processed_fields)
```

Comment:
Just a convention thing ... for a module exclusive "helper" method add a `private` section and move the `cleanup!` method to it (below it).

Reply:
Got it, thanks.